tests/testthat/test-as_curve_params.R

```diff
@@ -0,0 +1,20 @@
+test_that("`as_curve_params()` produces an error when non-curve data is provided", {
+  library(magrittr)
+  expect_error(
+    object = curve_data <-
+      "https://osf.io/download//n6cp3/" %>% # pop data
+      readr::read_rds() %>%
+      as_curve_params(),
+    regexp = "Please provide curve data",
+    fixed = TRUE
+  )
+})
+
+test_that("`as_curve_params()` produces expected results", {
+  expect_snapshot(
+    curve_data <-
+      "https://osf.io/download/rtw5k/" %>% # curve data
+      readr::read_rds() %>%
+      as_curve_params()
```

Comment:
if you look at the corresponding [snapshot file in this PR](https://github.com/UCD-SERG/serocalculator/blob/709aeb0f1d6ebdb9de4d9bc9e68aa98d9b88b2aa/tests/testthat/_snaps/as_curve_params.md?plain=1), you'll see that there's no output there, which means it isn't an effective test.
`expect_snapshot()` captures console output, and the code in lines 15-18 don't produce any, because the object was assigned to a variable name.
Please make two changes:

- [ ]  remove `curve_data <-` on line 15 so the printed console output gets captured correctly
- [ ] also, capture the full returned value using [`expect_snapshot_value()`](https://testthat.r-lib.org/reference/expect_snapshot_value.html)

Reply:
@d-morrison I am getting the error below when using `expect_snapshot_value`. 
<img width="482" alt="image" src="https://github.com/user-attachments/assets/40e4e6b7-b51b-401c-bff0-0b5e49b772a6"> . This is a reported bug on the function as reported [here](https://github.com/r-lib/testthat/issues/1678). It seems the bug has not been resolved.

Reply:
@chrisorwa great work finding the bug report; until that bug gets fixed, let's use `expect_snapshot_value(
style = "serialize")` as a workaround when this comes up. Let's still default to "deparse" when possible though, and only use "serialize" when "deparse" fails.